app-creator/client/src/redux/actions.ts

```diff
@@ -77,6 +79,23 @@ export const updateWidgetMetaData = (
   };
 };
 
+/**
+ * Sets the shared attribute on the widget object. The shared attribute
+ * lets us know if a widget is on the template or on the scratch panel.
+ */
+export const updateWidgetSharedStatus = (
```

Comment:
The word "shared" isn't clear whether it means that the widget is in a template or scratch space. I think of it more like "active" and "inactive", where an active widget is being used in a template, and an inactive widget is in the scratch space. There might be better adjectives!

Reply:
I'll add a TODO for now